### ISSUE TITLE
Best model calculation in classification training

### ISSUE THREAD

**Issue by zakenobi** [ultralytics/ultralytics#4061]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

Hi, when using the classification training I noticed the to accuracy metrics, `top1_acc` and `top5_acc`, I find them really useful but I observed that only one of them is taken into account when selecting the "best model", the `top5_acc`. For exemple I saw this with early stopping that would trigger only according to the top5_acc and most importantly at the end of the training it will save the model from the last epoch with the best top5 without considering the top1: 
![image](https://github.com/ultralytics/ultralytics/assets/54433310/66f1bd88-2fa7-46e3-a511-28edf05952a9)

In this case the best model was considered at epoch 25 but arguably I would say the best one was on epoch 19 if we consider both top1 and top5 accuracy

### Environment

- Ultralytics YOLOv8.0.136 🚀 Python-3.8.10 torch-2.0.1+cu117 CUDA:0 (Quadro RTX 4000, 7981MiB)
- Ubuntu 20.04
- Python 3.8.10

### Minimal Reproducible Example

Training params:
```model.train(data=os.path.join(current_path, data_path), epochs=40, imgsz=640, patience=5, dropout=0.2, batch=-1)```

Not sure if this is specific to my dataset but it is reproduced every time

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@zakenobi great news 😃! Your original issue may now be resolved ✅ a while ago. Classification fitness is computed as the mean of the two classification metrics.

https://github.com/ultralytics/ultralytics/blob/379f31904a2c2e0c88849df0cc56fb6dd0936e0f/ultralytics/utils/metrics.py#L964-L968

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by zakenobi** [comment]
Yes Indeed thanks! It's a little hard to keep up with all the fast changes ha ha

**Comment by shubham-kumar-ucentrics** [comment]
How do I use a different metric to select the best model ? Also, the shown values are computed over train or validation dataset ?

**Comment by glenn-jocher** [comment]
To use a different metric for selecting the best model, you may modify the `compute_fitness()` function in the `ultralytics/utils/metrics.py` file to align with your preferred metric. The shown values, such as `top1` and `top5` accuracy, are computed on the validation dataset. For more on YOLO11 performance metrics, visit [YOLO Performance Metrics](https://docs.ultralytics.com/guides/yolo-performance-metrics/).
